pkg/pool-linear/contracts/ampl/WAmplWAaveAmplLinearPool.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "../interfaces/IButtonWrapper.sol";
+import "../interfaces/IAToken.sol";
+import "../LinearPool.sol";
+
+/**
+ * @title WAmplWAaveAmplLinearPool
+ * @author @aalavandhan1984 (eng@fragments.org)
+ * @notice This linear pool is between wAMPL (wrapped AMPL)
+ *         and wAaveAMPL (wrapped aaveAMPL, or unbuttoned aAMPL).
+ * @dev The exchange rate between both is calculated based on:
+ *        - the rate between wAMPL and AMPL
+ *        - the rate between AMPL and aaveAMPL
+ *        - the rate between wAaveAMPL and aaveAMPL
+ *
+ *      Both AMPL and aaveAMPL are rebasing assets and are wrapped into a non-rebasing version
+ *      using warappers which implement the IButtonWrapper interface.
+ *      https://github.com/buttonwood-protocol/button-wrappers/blob/main/contracts/interfaces/IButtonWrapper.sol
+ */
+contract WAmplWAaveAmplLinearPool is LinearPool {
```

Comment:
Wondering if we should just call this ButtonAaveLinearPool to keep it general. Both sides are essentially ButtonTokens.

As far as I know, this implementation would work equivalently with aETH/btnETH for example. Once the aToken is Buttoned, it doesn't really matter if the aToken's underlying is AMPL or not. Same with WAMPL.

Reply:
you mean "un-buttoned" ?

Reply:
Ah, right-- UnButtonedAaveLinearPool

Reply:
👍